interp/dumpglob.mli

```diff
@@ -8,6 +8,95 @@
+   from a Coq file.
+
+   The [.glob] file format is notably used by [coqdoc] and [coq2html]
+   to generate links and other documentation meta-data.
+
+   Note that we consider this format a legacy one, and no stability
+   guarantees are provided as of today, as we search to replace this
+   format with a more structured and strongly-typed API.
+
+   However, we do provide up to date documentation about the format of
+   [.glob] files below.
+
+*)
+
+(** {2 The [.glob] file format}
+
+   [.glob] files contain a header, and then a list of entries, with
+  one line per entry.
+
+  {3 [.glob] header }
+
+  The header consists of two lines:
+
+[DIGEST: %md5sum_of_file]
+[%modpath]
+
+  where %modpath is the fully-qualified module name of the library that the
+  [.glob] file refers to. [%md5sum_of_file] may be NO if [-dump-glob file] was used.
+
+ {3 [.glob] entries }
+
+  There are 2 kinds of [.glob] entries:
+
+  - *definitions*: these entries correspond to definitions of inductives,
+    functions, binders, notations. They are written as:
+
+    [%kind %bc:%ec %secpath %name]
+
+    where [%kind] is one of
+    [{def,coe,subclass,canonstruc,ex,scheme,proj,inst,meth,defax,prfax,thm,prim,class,var,indrec,rec,corec,ind,variant,coind,constr,not,binder}],
+    meaning:
+    + [def] Definition
+    + [coe] Coertion
+    + [thm] Theorem
+    + [subclass] Sub Class
+    + [canonstruc] Canonical Declaration
+    + [ex] Example
+    + [scheme] Scheme
+    + [class] Class declaration
+    + [proj] Projection from a structure
+    + [inst] Instance
+    + [meth] Class Method
+    + [defax] Definitional assumption
+    + [prfax] Logical assumption
+    + [prim] Primitive
+    + [var] Variable reference
+    + [indrec] Inductive
+    + [rec] Inductive  (variant)
+    + [corec] Coinductive
+    + [ind] Record
+    + [variant] Record (variant)
+    + [coind] Coinductive Record
+    + [constr] Constructor
+    + [not] Notation
+    + [binder] Binder
+
+    [%bc] and [%ec] are respectively the start and end character for the file, [%secpath] the section path
+    (or [<>] if no section path) and [%name] the name of the defined object, or also [<>] in where no name applies.
+
+    Section paths are ...
+
+    + In the case of notations, [%name] is encoded as [:entry:scope:notation_string] where [_] is used to replace
+      spaces in the notation key, [%entry] is left empty if the notation entry is [constr],
+      and similarly [%scope] is empty if the corresponding notation has no associated scope.
+
+    + For binding variables, [:number] is added to distinguish uniquely different binding variables of the same name in a file.
+
+  - *references*: which identify the object a particular document piece of text points to;
+    their format is:
+
+    [R%%bc:%ec %filepath %secpath %name %kind]
+
+    where [%bc], [%ec], [%name], and [%kind] are as the above; [%filepath] contains the
+    absolute module path the object that is references lives in.
```

Comment:
This is a bit misleading.  It's not the absolute module path, but only the dirpath part of it, i.e., the absolute module path of the file the reference lives in.  (Also "that is references lives" is incorrect grammar here.). The rest of the module path (any non-file non-directory modules) are in the prefix of name.